### PR TITLE
Chart annotations

### LINKED ISSUE #3699: Allow for Annotations and Labels to be injected into helm chart templates
It would be great if we could inject labels and/or annotations into some of the templates in the helm chart. this would be particularly helpful for the CRD which is quite large, and ArgoCD doesn't like the size of the metadata (particularly `last-applied-configuration`). We could inject an argo sync annotation to help avoid that issue.

### PR DESCRIPTION
**Pull request by mwm5945**
**What this PR does / why we need it**: Adds configurable annotations to the CRDs and Manager deployment templates.

**Which issue(s) this PR fixes**:

Fixes #3699

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add annotation fields to helm template for CRD and deployment
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @mwm5945. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jklaise**
You can assign the PR to them by writing `/assign @jklaise` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by cliveseldon** [comment]
@mwm5945 We will probably need to change https://github.com/SeldonIO/seldon-core/blob/master/operator/helm/split_resources.py rather than the template files directly as they are generated by this script from the kustomize.

**Comment by mwm5945** [comment]
/assign @cliveseldon 

@cliveseldon updated :)

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@mwm5945: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 24c85eea677d361e0e987fdf84155b4beb22d786 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3724-notebooks-dffx8) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@mwm5945: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 24c85eea677d361e0e987fdf84155b4beb22d786 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3724-notebooks-kbs7b) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@mwm5945: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 24c85eea677d361e0e987fdf84155b4beb22d786 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3724-notebooks-4kxsq) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by cliveseldon** [comment]
Thanks @mwm5945 !

File: helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
   template:
     metadata:
       annotations:
+{{- with .Values.manager.annotations }}
+{{- toYaml . | nindent 4}}
```
Comment (agrski) [inline helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml:25]: :question: Couple of days late, but shouldn't this be an indent 8 spaces not 4?

Comment (mwm5945) [reply helm-charts/seldon-core-operator/templates/deployment_seldon-controller-manager.yaml:25]: yep--i'll fix in another pr. Good catch!

**Pull request merged**